website/content/docs/rfcs/web-ui-modernization.mdx

```diff
@@ -0,0 +1,137 @@
+---
+sidebar_label: Web UI modernization
+description: |-
+  Proposal to rewrite OpenBao's Web UI from EmberJS to React with TypeScript,
+introducing a modern, extensible, secure interface with plugin architecture,
+observability dashboards, and integration hub to revitalize community contributions
+and enhance user/operator experience.
+---
+
+
+
+# RFC - Web UI Modernization
+
+## Summary
+
+**OpenBao’s** current Web UI — based on **EmberJS** — is outdated and limits community contributions and modern feature development. This RFC proposes a complete rewrite using **React with TypeScript** to build a modern, secure, and extensible interface. The new UI will be developed as a separate application, introducing a secure **plugin architecture**, robust **observability dashboards**, and an **integration hub**. This initiative aims to revitalize community involvement, improve the user and operator experience, and establish a strong foundation for future innovation.
+
+## Problem Statement
+
+- **Obsolete Frontend:** EmberJS lacks community momentum; is hard to attract contributors and maintain.
+
+- **No Plugin Support:** The UI can’t be extended dynamically for operator or user needs.
+
+- **Clunky Permissions UX:** Users see, but can’t use, features; operators can’t easily hide UI elements based on roles/policies.
+
+- **Confusing URL Pathing:** Users often copy/paste UI URLs directly into the CLI, which fails due to mismatched endpoint paths, causing frustration and inconsistent outcomes.
```

Comment:
Not only this paths are being used by CLI, but in my experience it's also quiet difficult to built policies around UI & API usage alike. Some UI stuff needs other paths access than what path we use through the CLI. I would very much welcome a streamlining of the URL Paths.